pkg/webui/console/views/device-mac-settings/device-mac-settings.js

```diff
@@ -0,0 +1,244 @@
+      } catch (error) {
+        setError(error)
+        setSubmitting(false)
+      }
+    },
+    [appId, devId, device, isClassB, updateDevice, activationMode],
+  )
+
+  return (
+    <Container>
+      <IntlHelmet title={sharedMessages.macSettings} />
+      <Row>
+        <Col lg={8} md={12}>
+          <Form
+            horizontal
+            validateOnChange
+            enableReinitialize
+            error={error}
+            initialValues={initialValues}
+            validationSchema={validationSchema}
+            onSubmit={handleSubmit}
+          >
+            <Message component="h4" content={m.setMacSettings} />
+            {isABP && (
+              <>
+                <Form.Field
+                  title={m.rx1DelayTitle}
+                  description={m.rx1DelayDescription}
+                  name="mac_settings.rx1_delay.value"
+                  component={Select}
+                  options={rx1DelayOptions}
+                />
+                <Form.Field
+                  type="number"
+                  min={0}
+                  max={7}
+                  step={1}
+                  title={m.rx1DataRateOffsetTitle}
+                  name="mac_settings.rx1_data_rate_offset"
+                  component={Input}
+                />
+                <Form.Field
+                  title={m.resetsFCnt}
+                  onChange={handleResetsFCntChange}
+                  warning={resetsFCnt ? m.resetWarning : undefined}
+                  name="mac_settings.resets_f_cnt"
+                  component={Checkbox}
+                />
+              </>
+            )}
+            <Form.Field
+              title={m.rx2DataRateIndexTitle}
+              description={m.rx2DataDateIndexDescription}
+              name="mac_settings.rx2_data_rate_index.value"
+              component={Select}
+              options={dataRateIndexOptions}
+            />
+            <Form.Field
+              type="number"
+              min={100000}
+              step={1}
+              title={m.rx2FrequencyTitle}
+              description={m.rx2FrequencyDescription}
+              placeholder={m.frequencyPlaceholder}
+              name="mac_settings.rx2_frequency"
+              component={Input}
+            />
+            <Form.Field
+              indexAsKey
+              name="mac_settings.factory_preset_frequencies"
+              component={KeyValueMap}
+              title={m.factoryPresetFreqTitle}
+              description={m.factoryPresetFreqDescription}
+              addMessage={m.freqAdd}
+              valuePlaceholder={m.frequencyPlaceholder}
+            />
+            {isClassB && (
+              <Form.Field
+                title={m.pingSlotPeriodicityTitle}
+                description={m.pingSlotPeriodicityDescription}
+                name="mac_settings.ping_slot_periodicity.value"
+                component={Select}
+                options={pingSlotPeriodicityOptions}
+              />
+            )}
```

Comment:
`mac_settings.ping_slot_frequency.value` must be available as well (otherwise class B devices cannot be created in `AU_915_928`, for example)

Reply:
`ping_slot_periodicity` field is shown and validated only for devices with `supports_class_b = true`. not sure what you mean here

Reply:
Please reread my comment. I'm talking about the frequency here.

Reply:
added